internal/apps/cmp/impl/clusters/add_edge_cluster.go

```diff
@@ -244,19 +249,20 @@ func (c *Clusters) MonitorCloudCluster() (abort bool, err error) {
 			_ = c.processFailedPipeline(ctx, record, err)
 			continue
 		}
-		if len(dto.PipelineStages) == 0 {
+		if len(dto.Data.PipelineStages) == 0 {
 			err = fmt.Errorf("empty pipeline stages, pipelineid: %d", record.PipelineID)
 			_ = c.processFailedPipeline(ctx, record, err)
 			continue
 		}
-		for _, stage := range dto.PipelineStages {
+		for _, stage := range dto.Data.PipelineStages {
 			if len(stage.PipelineTasks) == 0 {
 				err = fmt.Errorf("empty task in pipeline stage")
 				_ = c.processFailedPipeline(ctx, record, err)
 				break
 			}
 			for _, task := range stage.PipelineTasks {
-				if task.Status.IsFailedStatus() {
+				taskStatus := apistructs.PipelineStatus(task.Status)
```

Comment:
move pipeline status into one of pipeline provider.

Reply:
apistructs of pipeline will be moved in next pr